ca_qc_saint_jean_sur_richelieu/people.py

```diff
@@ -1,75 +1,43 @@
+            name = councillor.xpath('.//a//text()')[0]
+            url = councillor.xpath('.//a//@href')[0]
 
-            contacts = page.xpath('.//td[@class="ms-rteTableOddCol-0"]//text()')
-            for contact in contacts:
-                if re.findall(r'[0-9]{4}', contact):
-                    phone = contact.strip().replace(' ', '-')
-                    p.add_contact('voice', phone, 'legislature')
-            get_links(p, page.xpath('.//td[@class="ms-rteTableOddCol-0"]')[0])
+            if (councillor.xpath('.//a/@href[contains(.,"maire")]')):
+                ward = 'Saint-Jean-sur-Richelieu'
+                role = 'Maire'
+            else:
+                role = 'Conseiller'
+                district = councillor.xpath('.//p[contains(.,"District")]/text()')
+                for i in district:
+                    sep = ','
+                    ward = i.split(sep, 1)[0]
 
-            email = self.get_email(page)
-            p.add_contact('email', email)
-            yield p
+            yield self.councillor_data(url, name, ward, role)
 
-    def scrape_mayor(self, div):
-        name = div.xpath('.//a')[0].text_content()
-        url = div.xpath('.//a/@href')[0]
-        page = self.lxmlize(url)
-        contact_url = page.xpath('//a[@title="Joindre le maire"]/@href')[0]
-        contact_page = self.lxmlize(contact_url)
+    def councillor_data(self, url, name, ward, role):
+        node = self.lxmlize(url)
+        photo_url_rel = node.xpath('//div[@class="fl-photo-content fl-photo-img-jpg"]//img/@src')[0]
+        photo_url = urljoin(url, photo_url_rel)
 
-        p = Person(primary_org='legislature', name=name, district='Saint-Jean-sur-Richelieu', role='Maire')
+        p = Person(primary_org='legislature', name=name, district=ward, role=role)
         p.add_source(COUNCIL_PAGE)
         p.add_source(url)
-        p.add_source(contact_url)
-
-        p.image = div.xpath('./preceding-sibling::td//img/@src')[-1]
-
-        contacts = contact_page.xpath('//div[@id="ctl00_PlaceHolderMain_ctl01_ctl01__ControlWrapper_RichHtmlField"]//div/font/text()')
-        address = ' '.join(contacts[:4])
-        phone = contacts[-3].split(':')[1].strip().replace(' ', '-')
-        fax = contacts[-2].split(':')[1].strip().replace(' ', '-')
-        p.add_contact('address', address, 'legislature')
-        p.add_contact('voice', phone, 'legislature')
-        p.add_contact('fax', fax, 'legislature')
-        # mayor's email is a form
-        return p
-
+        p.image = photo_url
 
-def get_links(councillor, div):
-    links = div.xpath('.//a')
-    for link in links:
-        link = link.attrib['href']
+        if node.xpath('//div[contains(@class,"fl-module-content fl-node-content")]/div[@class="fl-rich-text"]/p[2]/strong//text()'):
+            p.add_contact('voice', node.xpath('//div[contains(@class,"fl-module-content fl-node-content")]/div[@class="fl-rich-text"]/p[2]/strong//text()')[0], 'legislature')
```

Comment:
Instead of repeating the XPath (which is harder to maintain), do:

```python
voice = node.xpath(…)
if voice:
    p.add_contact('voice', voice[0], 'legislature')
```